spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -425,7 +431,96 @@ public String branch() {
+    writeProperties.putAll(dataWriteProperties());
+    writeProperties.putAll(deleteWriteProperties());
+    return writeProperties;
+  }
+
+  private Map<String, String> dataWriteProperties() {
+    Map<String, String> writeProperties = Maps.newHashMap();
+
+    FileFormat dataFormat = dataFileFormat();
+    switch (dataFormat) {
+      case PARQUET:
+        writeProperties.put(PARQUET_COMPRESSION, parquetCompressionCodec());
+        String parquetCompressionLevel = parquetCompressionLevel();
+        if (parquetCompressionLevel != null) {
+          writeProperties.put(PARQUET_COMPRESSION_LEVEL, parquetCompressionLevel);
+        }
+        break;
+
+      case AVRO:
+        writeProperties.put(AVRO_COMPRESSION, avroCompressionCodec());
+        String avroCompressionLevel = avroCompressionLevel();
+        if (avroCompressionLevel != null) {
+          writeProperties.put(AVRO_COMPRESSION_LEVEL, avroCompressionLevel);
+        }
+        break;
+
+      case ORC:
+        writeProperties.put(ORC_COMPRESSION, orcCompressionCodec());
+        writeProperties.put(ORC_COMPRESSION_STRATEGY, orcCompressionStrategy());
+        break;
+
+      default:
+        // skip
+    }
+
+    return writeProperties;
+  }
+
+  private Map<String, String> deleteWriteProperties() {
+    Map<String, String> writeProperties = Maps.newHashMap();
+
+    FileFormat deleteFormat = deleteFileFormat();
+    switch (deleteFormat) {
+      case PARQUET:
+        String parquetCodec = deleteParquetCompressionCodec();
+        writeProperties.put(
+            DELETE_PARQUET_COMPRESSION,
+            parquetCodec != null ? parquetCodec : parquetCompressionCodec());
+        String deleteParquetCompressionLevel = deleteParquetCompressionLevel();
+        if (deleteParquetCompressionLevel != null) {
+          writeProperties.put(DELETE_PARQUET_COMPRESSION_LEVEL, deleteParquetCompressionLevel);
+        } else if (parquetCompressionLevel() != null) {
+          writeProperties.put(DELETE_PARQUET_COMPRESSION_LEVEL, parquetCompressionLevel());
+        }
+        break;
+
+      case AVRO:
+        String avroCodec = deleteAvroCompressionCodec();
+        writeProperties.put(
+            DELETE_AVRO_COMPRESSION, avroCodec != null ? avroCodec : avroCompressionCodec());
+        String deleteAvroCompressionLevel = deleteAvroCompressionLevel();
+        if (deleteAvroCompressionLevel != null) {
+          writeProperties.put(DELETE_AVRO_COMPRESSION_LEVEL, deleteAvroCompressionLevel);
+        } else if (avroCompressionLevel() != null) {
+          writeProperties.put(DELETE_AVRO_COMPRESSION_LEVEL, avroCompressionLevel());
+        }
+        break;
+
+      case ORC:
+        String orcCodec = deleteOrcCompressionCodec();
+        writeProperties.put(
+            DELETE_ORC_COMPRESSION, orcCodec != null ? orcCodec : orcCompressionCodec());
+        String strategy = deleteOrcCompressionStrategy();
+        if (strategy != null) {
```

Comment:
Simlarly, maybe setWithFallback(writeProperties, DELETE_ORC_COMPRESSION_STRATEGY, deleteOrcCompressionStrategy(), orcCompressionStrategy())  (I feel null handling is similar in the two cases?)

Reply:
Thanks. Fixed.